### PR TITLE
Refactor DeepSeek for vLLM 0.26 GPU remote experts

### PR DESCRIPTION
**Pull request by jiaran-king**
# Draft: support vLLM 0.26 GPU V1 and remote MoE experts boundary

> Draft PR #176 targets `main`. The reviewer-driven simplification is included
> in the squashed head `99643a9`.

## Summary

This Draft advances the AFD GPU backend to exact vLLM `0.26.0` and moves
DeepSeek MoE communication from the whole-MLP boundary to a remote-experts
boundary.

It:

- uses GPU ModelRunner V1 (`VLLM_USE_V2_MODEL_RUNNER=0`);
- keeps native `Model.forward`, `DecoderLayer.forward`, and
  `DeepseekV2MoE.forward`;
- adds a parameter-free Attention experts proxy;
- keeps native FFN MoE runners as the only owners of routed/shared experts;
- supports both Attention-side and FFN-side gate placement;
- reuses the existing connector data-path methods instead of adding an
  experts-specific public connector API;
- sends optional Attention-side router logits in order after hidden states
  through the existing P2P custom ops and ordering token;
- aligns DBO and CUDA Graph execution with vLLM `0.26.0`.

### Attention-side gate

```text
native DeepseekV2MoE.forward
→ native gate
→ parameter-free AFD experts proxy
→ existing send_attn_output(hidden states, optional router logits)
→ native FFN external-router MoE
→ existing send_ffn_output(final hidden states)
```

### FFN-side gate

```text
native DeepseekV2MoE.forward
→ parameter-free internal-router proxy
→ existing send_attn_output(hidden states)
→ native FFN gate + experts
→ existing send_ffn_output(final hidden states)
```

Dense layers retain the existing whole-MLP remote path.

## Simplification

The revised implementation:

- leaves `AFDConnectorBase` unchanged relative to `main`;
- removes `send_expert_input`, `recv_expert_input`, `send_expert_output`, and
  `recv_expert_output`;
- removes request/response headers, transfer IDs, router dtype codes, and
  grouped P2P custom ops;
- keeps routing placement/shape as a model-owned local contract;
- reduces the final `p2p.py` production diff against `main` to `+108/-10`;
- removes the standalone remote-experts module, handle/state machine, protocol,
  and stateless executor;
- reuses the existing `RemoteFFNProxy` transfer path and calls the native model
  interface directly from the FFN runner;
- uses native `PPMissingLayer` instead of AFD-only role placeholders.

The connector simplification removed 1,210 lines relative to the earlier Draft.
The final remote-experts cleanup removed another 440 lines from the working diff
(8 files, 130 insertions and 570 deletions).

## Scope

In scope:

- CUDA GPU backend;
- exact vLLM `0.26.0`;
- ModelRunner V1;
- DeepSeek-V2-Lite hardware validation;
- both gate placements;
- eager, CUDA Graph, TP2, DP2/EP2, and representative DBO paths.

Out of scope:

- NPU/Ascend hardware validation;
- ModelRunner V2;
- quantized MoE and EPLB validation;
- full 1,319-sample GSM8K reporting.

## Validation

Current squashed-head validation:

- focused unit suite: 92 passed;
- complete unit suite: 353 passed, 29 skipped;
- Ruff, format, and diff checks: passed;
- targeted GPU paths passed:
  - `1A1F` FFN-side gate eager;
  - `1A1F` Attention-side gate eager;
  - TP2/EP2 CUDA Graph (`1 passed in 168.52 s`).

Earlier migration and connector-simplification evidence also includes:

- `2A2F` Attention-side gate eager;
- `2A2F` Attention-side gate DBO + CUDA Graph with 8/8 matching completions;
- focused numerical oracles for both gate placements;
- real-checkpoint loader and native-runner ownership checks;
- GSM8K `50/1319` smoke with 50/50 HTTP successes and strict/flexible exact
  match `0.26`.

The current-head tests target the transfer path changed by the final cleanup.
The complete 20-node repository GPU matrix has not been rerun against this exact
head.

## Known Draft limitations

- The external-router FFN model currently retains a dormant native gate to
  preserve the native model tree and loader contract. Gate compute remains on
  Attention.
- The accepted FFN receive-loop/connector-close shutdown race can emit a
  post-request shutdown error, although focused runs release owned resources.
- The repository-wide GPU E2E release gate is not complete.
- NPU, ModelRunner V2, quantization, and EPLB are not claimed.

## Reviewer focus

Please focus review on:

1. native-versus-AFD ownership at the experts boundary;
2. dual gate-placement construction and loading;
3. reuse of the existing four connector data-path methods;
4. ordered optional router-logits transfer under eager, Graph, TP/EP, and DBO;
5. vLLM `0.26.0` runtime patch scope and exact signatures.

## Checklist

- [x] Directly based on current `main`
- [x] Exact vLLM `0.26.0`
- [x] GPU ModelRunner V1 only
- [x] Both gate placements implemented
- [x] No new experts methods in `AFDConnectorBase`
- [x] No second experts request/response wire protocol
- [x] Remove unused remote-experts abstractions and duplicate tests
- [x] Focused eager/Graph/TP/DP/EP/DBO evidence
- [x] GSM8K 50-sample accuracy smoke
- [x] Squash and push the reviewed implementation
- [ ] Full 20-test GPU E2E accounting, if required before Ready

### CODE REVIEW

**Review by jiangkuaixue123 — COMMENTED** [review]
p2p.py should not require this many code changes, and base.py should not introduce any new interfaces. Please simplify the implementation; the scope of this change is too large.

**Comment by jiaran-king** [comment]
> p2p.py should not require this many code changes, and base.py should not introduce any new interfaces. Please simplify the implementation; the scope of this change is too large.

thanks, agreed — I simplified the implementation and pushed the update in 178beb3.
-AFDConnectorBase now has no diff from main; the four experts-specific connector interfaces were removed.
-The experts path now reuses the existing send_attn_output / recv_attn_output and send_ffn_output / recv_ffn_output methods.
-Removed the separate request/response headers, transfer IDs, grouped P2P custom ops, and the second wire protocol.
-Attention-side routing now only transfers an optional ordered router_logits tensor through the existing P2P operation and ordering token.
-The resulting p2p.py diff is +108/-10 versus main; this revision removes 1,210 lines from the previous Draft.
The simplification preserves native MoE forward behavior and native FFN runner ownership. The unit suite passed (356 passed, 29 skipped), along with all five focused GPU cases covering both gate placements, TP2/EP2 Graph, 2A2F, and DBO+Graph.
Please take another look.

File: afd_plugin/v1/worker/dbo.py [hunk 1]
```diff
@@ -29,19 +30,18 @@ def register_dbo_yield_custom_op() -> None:
     if _AFD_DBO_YIELD_OP_REGISTERED:
         return
 
-    def afd_manual_dbo_yield_op(x: torch.Tensor) -> torch.Tensor:
+    def afd_manual_dbo_yield_op(x: torch.Tensor) -> None:
```
Comment (jiangkuaixue123) [inline afd_plugin/v1/worker/dbo.py:33]: 为什么要修改这个return
